docs(rust): mention ADBC_SNOWFLAKE_GO_LIB_DIR env var in README

### What feature or improvement would you like to see?

There is no mention of `ADBC_SNOWFLAKE_GO_LIB_DIR` env var.
It appears that we need it to build and test.
It would be helpful to have a note saying that we need to set `ADBC_SNOWFLAKE_GO_LIB_DIR` to build and test.

https://github.com/apache/arrow-adbc/blob/c08fdfca8e99ff86decd00f2936c14c71c25fcf1/rust/README.md#L33-L41

```console
$ LD_LIBRARY_PATH="/path/to/lib:$LD_LIBRARY_PATH" cargo test --all-features --workspace
   Compiling adbc_snowflake v0.19.0 (/home/daipom/work/apache-arrow/arrow-adbc/rust/driver/snowflake)
error: linking with `cc` failed: exit status: 1
  |
  = note:  "cc" "-m64" "/tmp/rustcAcxIWp/symbols.o" "<122 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "-lsnowflake" "-Wl,-Bdynamic" "-lresolv" "-ladbc_driver_snowflake" "-Wl,-Bstatic" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/deps/{libdotenvy-d971bd2ac71104e2.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libtest-*,libgetopts-*,libunicode_width-*,librustc_std_workspace_std-*}.rlib" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/deps/{libregex-054267695ddcc422.rlib,libregex_automata-e178f9689d5d73b1.rlib,libaho_corasick-96de60b9858755f6.rlib,libmemchr-df7138072aead54d.rlib,libregex_syntax-e9911f1b4a36081d.rlib,liburl-a450b8ba41512ff0.rlib,libidna-1124dbc57fd466e2.rlib,libidna_adapter-5a4c6b92bb2a73d5.rlib,libicu_normalizer-73ec3a34247c9de9.rlib,libicu_normalizer_data-5d1a678d344b291c.rlib,libwrite16-4a21213400bb3552.rlib,libutf8_iter-4dd926676127a369.rlib,libutf16_iter-c1d656f0774d7f77.rlib,libsmallvec-5604f406424ccf52.rlib,libicu_properties-b022a9239de993bd.rlib,libicu_properties_data-2325023fb5c18f26.rlib,libicu_locid_transform-b0968e7859f9760c.rlib,libicu_locid_transform_data-8ad0c4a60b07d45b.rlib,libicu_collections-219ce6b73e2e1b7e.rlib,libicu_provider-ef7265ae831fedd1.rlib,libicu_locid-311b5c9804b2609e.rlib,liblitemap-4182afdfcf84e15d.rlib,libtinystr-d904807504123793.rlib,libzerovec-1cdfdc33fe019083.rlib,libwriteable-db6a66dd17f3debd.rlib,libyoke-f7180c788f2ae119.rlib,libzerofrom-15d88d4508209965.rlib,libstable_deref_trait-567eccf8a716e5cb.rlib,libform_urlencoded-25d070486319ce04.rlib,libpercent_encoding-3c9d9c63ad89d268.rlib,libadbc_core-fb1fbda3a169b49b.rlib,liblibloading-bc259c0c6e3a10f4.rlib,libarrow_array-246d2bdb8102d99b.rlib,libchrono_tz-896e5af71b780723.rlib,libphf-c11f112d778510f8.rlib,libphf_shared-fd70ad01fa6cd192.rlib,libsiphasher-08157d4b848ddf53.rlib,libahash-20a3a44cad151294.rlib,libgetrandom-56881cb6f3ec945c.rlib,liblibc-22799f3321850230.rlib,libonce_cell-c71ece0e52800bc8.rlib,libzerocopy-597f3130f2000d61.rlib,libbyteorder-42e03b0443b70f8a.rlib,libhashbrown-680ed0a3438242a6.rlib,libfoldhash-96860c9f2ddf0951.rlib,libequivalent-82253b26ca019f5d.rlib,liballocator_api2-49075a3081dff7a6.rlib,libchrono-95f76526792f5616.rlib,libiana_time_zone-88bad21c3dddeda6.rlib,libarrow_data-1e8860b81bc2d337.rlib,libarrow_schema-7507a7cde0ece797.rlib,libbitflags-a31b82e9ca12bca2.rlib,libarrow_buffer-b30d5f0b92eafdb4.rlib,libbytes-23ef84c3ede8db87.rlib,libhalf-521ca67a721daad1.rlib,libcfg_if-f52ed1292e79c10c.rlib,libnum-e5d8582d3055c328.rlib,libnum_iter-49a0f47daaf5969e.rlib,libnum_rational-ec7b0c650d4f28c3.rlib,libnum_complex-1ba1c8a287f776c1.rlib,libnum_bigint-119b6046dd8a14b2.rlib,libnum_integer-112ffa6290f2b8c9.rlib,libnum_traits-312d161a8cb7f4e8.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-ldl" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/tmp/rustcAcxIWp/raw-dylibs" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/build/adbc_snowflake-d5805cbc7fda02a9/out" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/deps/adbc_snowflake-c10ef1ecb4318f98" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: /home/daipom/miniforge3/envs/adbc/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -ladbc_driver_snowflake: No such file or directory
          collect2: error: ld returned 1 exit status
          

error: could not compile `adbc_snowflake` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit status: 1
  |
  = note:  "cc" "-m64" "/tmp/rustczdvGhM/symbols.o" "<69 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libtest-*,libgetopts-*,libunicode_width-*,librustc_std_workspace_std-*}.rlib" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/deps/{libadbc_snowflake-39a2ee786d8df6d2.rlib,libdotenvy-d971bd2ac71104e2.rlib,libregex-054267695ddcc422.rlib,libregex_automata-e178f9689d5d73b1.rlib,libaho_corasick-96de60b9858755f6.rlib,libmemchr-df7138072aead54d.rlib,libregex_syntax-e9911f1b4a36081d.rlib,liburl-a450b8ba41512ff0.rlib,libidna-1124dbc57fd466e2.rlib,libidna_adapter-5a4c6b92bb2a73d5.rlib,libicu_normalizer-73ec3a34247c9de9.rlib,libicu_normalizer_data-5d1a678d344b291c.rlib,libwrite16-4a21213400bb3552.rlib,libutf8_iter-4dd926676127a369.rlib,libutf16_iter-c1d656f0774d7f77.rlib,libsmallvec-5604f406424ccf52.rlib,libicu_properties-b022a9239de993bd.rlib,libicu_properties_data-2325023fb5c18f26.rlib,libicu_locid_transform-b0968e7859f9760c.rlib,libicu_locid_transform_data-8ad0c4a60b07d45b.rlib,libicu_collections-219ce6b73e2e1b7e.rlib,libicu_provider-ef7265ae831fedd1.rlib,libicu_locid-311b5c9804b2609e.rlib,liblitemap-4182afdfcf84e15d.rlib,libtinystr-d904807504123793.rlib,libzerovec-1cdfdc33fe019083.rlib,libwriteable-db6a66dd17f3debd.rlib,libyoke-f7180c788f2ae119.rlib,libzerofrom-15d88d4508209965.rlib,libstable_deref_trait-567eccf8a716e5cb.rlib,libform_urlencoded-25d070486319ce04.rlib,libpercent_encoding-3c9d9c63ad89d268.rlib,libadbc_core-fb1fbda3a169b49b.rlib,liblibloading-bc259c0c6e3a10f4.rlib,libarrow_array-246d2bdb8102d99b.rlib,libchrono_tz-896e5af71b780723.rlib,libphf-c11f112d778510f8.rlib,libphf_shared-fd70ad01fa6cd192.rlib,libsiphasher-08157d4b848ddf53.rlib,libahash-20a3a44cad151294.rlib,libgetrandom-56881cb6f3ec945c.rlib,liblibc-22799f3321850230.rlib,libonce_cell-c71ece0e52800bc8.rlib,libzerocopy-597f3130f2000d61.rlib,libbyteorder-42e03b0443b70f8a.rlib,libhashbrown-680ed0a3438242a6.rlib,libfoldhash-96860c9f2ddf0951.rlib,libequivalent-82253b26ca019f5d.rlib,liballocator_api2-49075a3081dff7a6.rlib,libchrono-95f76526792f5616.rlib,libiana_time_zone-88bad21c3dddeda6.rlib,libarrow_data-1e8860b81bc2d337.rlib,libarrow_schema-7507a7cde0ece797.rlib,libbitflags-a31b82e9ca12bca2.rlib,libarrow_buffer-b30d5f0b92eafdb4.rlib,libbytes-23ef84c3ede8db87.rlib,libhalf-521ca67a721daad1.rlib,libcfg_if-f52ed1292e79c10c.rlib,libnum-e5d8582d3055c328.rlib,libnum_iter-49a0f47daaf5969e.rlib,libnum_rational-ec7b0c650d4f28c3.rlib,libnum_complex-1ba1c8a287f776c1.rlib,libnum_bigint-119b6046dd8a14b2.rlib,libnum_integer-112ffa6290f2b8c9.rlib,libnum_traits-312d161a8cb7f4e8.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-lresolv" "-ladbc_driver_snowflake" "-ldl" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/tmp/rustczdvGhM/raw-dylibs" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/build/adbc_snowflake-d5805cbc7fda02a9/out" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/daipom/work/apache-arrow/arrow-adbc/rust/target/debug/deps/driver-aca81b540a3c3769" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: /home/daipom/miniforge3/envs/adbc/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -ladbc_driver_snowflake: No such file or directory
          collect2: error: ld returned 1 exit status
          

error: could not compile `adbc_snowflake` (test "driver") due to 1 previous error
```

```console
$ LD_LIBRARY_PATH="/path/to/lib:$LD_LIBRARY_PATH" ADBC_SNOWFLAKE_GO_LIB_DIR="/path/to/lib" cargo test --all-features --workspace
(All tests pass)
```

https://github.com/apache/arrow-adbc/blob/c08fdfca8e99ff86decd00f2936c14c71c25fcf1/.github/workflows/rust.yml#L120-L121

--- a/rust/README.md
+++ b/rust/README.md
@@ -36,9 +36,11 @@ To run the integration tests you must:
 
 1. Install a recent version of [protobuf](https://github.com/protocolbuffers/protobuf) and have its `bin` (`protoc`) in path.
 1. Install [SQLite](https://www.sqlite.org/) and have its dynamic library in path.
-1. Build the official ADBC SQLite driver by following the [documentation](../CONTRIBUTING.md).
+1. Build the official ADBC SQLite driver and ADBC Snowflake driver by following the [documentation](../CONTRIBUTING.md).
+   * Add `-DADBC_DRIVER_SNOWFLAKE=ON` option to build the Snowflake driver.
 1. Place the resulting object file into your dynamic loader path or set
    `LD_LIBRARY_PATH/DYLD_LIBRARY_PATH` appropriately.
+1. Set `ADBC_SNOWFLAKE_GO_LIB_DIR` appropriately.
 1. Run `cargo test --all-features --workspace`
 
 ## Writing Drivers

